src/operations/lightning.rs

```diff
@@ -198,57 +202,55 @@ pub async fn auth(creds: Credentials) -> Result<Tokens> {
 }
 
 /// Creates a lightning invoice
-pub async fn create_invoice(description: &str, amount: u64, token: &str) -> Result<String> {
+pub async fn create_invoice(description: String, amount: u64, token: String) -> Result<String> {
     let endpoint = LNDHUB_ENDPOINT.to_string();
     let url = format!("{endpoint}/addinvoice");
     let req = AddInvoiceReq {
-        memo: description.to_string(),
+        memo: description,
         amt: amount.to_string(),
     };
-    let response = post_json_auth(&url, &Some(req), Some(token)).await?;
+    let response = post_json_auth(&url, &Some(req), Some(&token)).await?;
     let invoice: AddInvoiceRes = serde_json::from_str(&response)?;
 
     Ok(invoice.payment_request)
 }
 
 /// Decode a lightning invoice (bolt11)
-pub async fn decode_invoice(invoice: &str, token: &str) -> Result<Invoice> {
+pub async fn decode_invoice(invoice: String, token: String) -> Result<Invoice> {
     let endpoint = LNDHUB_ENDPOINT.to_string();
     let url = format!("{endpoint}/decodeinvoice?invoice={invoice}");
-    let response = get(&url, Some(token)).await?;
+    let response = get(&url, Some(&token)).await?;
     let invoice: Invoice = serde_json::from_str(&response)?;
 
     Ok(invoice)
 }
 
 /// Get user lightning balance
-pub async fn get_balance(token: &str) -> Result<BalanceRes> {
+pub async fn get_balance(token: String) -> Result<BalanceRes> {
     let endpoint = LNDHUB_ENDPOINT.to_string();
     let url = format!("{endpoint}/balance");
-    let response = get(&url, Some(token)).await?;
+    let response = get(&url, Some(&token)).await?;
     let invoice: BalanceRes = serde_json::from_str(&response)?;
 
     Ok(invoice)
 }
 
 /// Pay a lightning invoice
-pub async fn pay_invoice(invoice: &str, token: &str) -> Result<PayInvoiceMessage> {
+pub async fn pay_invoice(invoice: String, token: String) -> Result<PayInvoiceMessage> {
     let endpoint = LNDHUB_ENDPOINT.to_string();
     let url = format!("{endpoint}/payinvoice");
-    let req = InvoiceReq {
-        invoice: invoice.to_string(),
-    };
-    let response = post_json_auth(&url, &Some(req), Some(token)).await?;
+    let req = InvoiceReq { invoice };
+    let response = post_json_auth(&url, &Some(req), Some(&token)).await?;
     let response: PayInvoiceMessage = serde_json::from_str(&response)?;
 
     Ok(response)
 }
 
 /// Get successful lightning transactions user made. Order newest to oldest.
-pub async fn get_txs(token: &str, limit: u32, offset: u32) -> Result<Vec<Tx>> {
```

Comment:
I'd prefer to keep the internal methods as string references, and then pass them in by reference in web.rs. Be sure to see how it's done in the other web.rs methods.

Reply:
done the String -> &str, there is something that I don't get it, why linters are not passing, I see they are broken but in files that are not related with this PR